internal/pkg/manifest/lb_web_svc.go

```diff
@@ -241,12 +241,13 @@ type NetworkLoadBalancerConfiguration struct {
 
 // NetworkLoadBalancerListener holds listener configuration for NLB.
 type NetworkLoadBalancerListener struct {
-	Port            *string            `yaml:"port"`
-	HealthCheck     NLBHealthCheckArgs `yaml:"healthcheck"`
-	TargetContainer *string            `yaml:"target_container"`
-	TargetPort      *int               `yaml:"target_port"`
-	SSLPolicy       *string            `yaml:"ssl_policy"`
-	Stickiness      *bool              `yaml:"stickiness"`
+	Port                *string            `yaml:"port"`
+	HealthCheck         NLBHealthCheckArgs `yaml:"healthcheck"`
+	TargetContainer     *string            `yaml:"target_container"`
+	TargetPort          *int               `yaml:"target_port"`
+	SSLPolicy           *string            `yaml:"ssl_policy"`
+	Stickiness          *bool              `yaml:"stickiness"`
+	DeregistrationDelay *time.Duration     `yaml:"deregistration_delay"`
```

Comment:
How discoverable is the deregistration_delay field in the backend service manifest package?  Is it possible that customers set one of these fields and expect it to set both of them, or is that considered a bug? Does that make sense?

I think as our manifest gets more complicated it makes sense for use to be careful to make sure that we have some way of surfacing fields that might get buried. +1 for intellisense schema

Reply:
If I am understanding your question correctly, then `deregistration_delay` is a target group level setting, so customers have to set that value for every listener in NLB and every listener rule in ALB. We support NLB only for LBWS for now, so coming back to your question of the backend service, we will only have ALB there, and we can set `deregistration_delay` on every listener rule of internal ALB. 